docs/misc/message_system.md

```diff
@@ -0,0 +1,242 @@
+# Message System
+
+Message system was implemented to allow sending emergency messages to Trezor Suite app to a user with specific stack.
+
+[Example messages](https://www.notion.so/1d622e7079c14f64a03fa17f50bce14f?v=b8649de184c3437abb983990c5513e5f)
+
+[Issue on Github](https://github.com/trezor/trezor-suite/issues/2752)
+
+## Types of in-app messages
+
+There are four ways of displaying message to a user.
+
+- banner
+   - looks like a cookie bar above the page
+- modal
+   - *TODO: missing implementation*
+- context 
+   - messages on specific places in app - high level (e.g. settings page)
+   - *TODO: missing implementation*
+- super-context 
+   - messages on specific places in the app - low level (e.g. category in settings page)
+   - *TODO: missing implementation*
+
+## Implementation
+
+### Config
+
+The system of messages is based on a configuration file in which messages with specific conditions ​are described. If specific conditions are satisfied, the message is shown to a user.
+
+Current configuration file is located in `packages/suite-data/src/message-system/config`. Its name is `config.vX.json`. The `X` express current version messaging system.
+
+The config is fetched at launch of the application and then every six hours (If the application is running). It remembers the previously fetched config to inform the user even if he is offline. For this reason, the latest available config during build time is bundled with the application.
+
+If fetching of a new config fails, the fetching process is repeated every 10 minutes.
+### Schema
+
+The configuration structure is specified in JSON file using JSON schema. The file can be found in `packages/suite-data/src/message-system/schema` folder. Its name is `config.schema.vX.json`.
+
+We use JSON schema for 2 reasons:
+- generating TypeScript types
+- validating configuration file
+
+### Types
+
+Types are generated from JSON-schema during the `build:libs` process or can be generated manually by `yarn workspace @trezor/suite-data msg-system-types`. A `messageSystem.ts` file is created in `packages/suite/src/types/suite`. 
+
+- This file should never be changed manually.
+- This file is committed into the repository. 
+
+### Signing
+
+To ensure the authenticity of a configuration file, JSON Web Signatures are used. The configuration file is signed by a private key using elliptic curves (ES256) and the data specified in the config are used. The authenticity is verified on client side using corresponding public key.
+
+### CI job
+
+#### Validation
+
+- Validation of configuration file is performed in CI job in `validation` phase. It is used to detect possible structure and semantic errors. 
```

Comment:
Does this validation include a check that the generated types in `messageSystem.ts` (as comitted) correspond to the JSON schema? Since we decided to track a generated file in git we should at least ensure it won't lag behind its source.

Reply:
I was thinking if it is good idea to commit types into repo as they are generated on `build:libs`, this is probably a reason, why it would be better not to do it.

Reply:
It's generally not the best practice to track generated files. Though sometimes simply commiting the file is simpler than ensuring it's automaticaly generated whenever needed.

In our case there's the `build:libs` job in the root package.json of the monorepo that is already required for suite to  build successfuly. (Right?) If we can ensure that `messageSystem.ts` is generated with this script too I'd suggest not to track it in git.

Reply:
It's generally not the best practice to track generated files. Though sometimes simply commiting the file is simpler than ensuring it's automaticaly generated whenever needed.

In our case, as you write, there's the `build:libs` job in the root package.json of the monorepo that is already required for suite to  build successfuly. And since `messageSystem.ts` is generated by that job too there's IMO no need to track it in git.

Reply:
Generated on ```build:libs``` now. Not commited to repo